`just build` is not idempotent — Nestor cmake re-run fails with find_package(httplib) error

## Summary
`just build` succeeds on a clean checkout but fails on re-runs against an existing `build/` directory. The `_build-nestor` cmake step cannot find `httplib-config.cmake` on re-run, making the build non-idempotent. This is a POLA violation.

## Reproduction
```bash
just build           # succeeds (first run, conan caches deps)
just build           # fails at Nestor cmake configure
# CMake Error at CMakeLists.txt:39 (find_package):
#   Could not find a package configuration file provided by 'httplib'
#   with any of the following names: httplibConfig.cmake, httplib-config.cmake.
#   Add the installation prefix of 'httplib' to CMAKE_PREFIX_PATH.
```

## Root Cause
The `_build-nestor` recipe passes `-DCMAKE_TOOLCHAIN_FILE` but the cmake configure step for Nestor does not re-run `conan install` to regenerate the toolchain when the build directory already exists. The conan-generated `httplib-config.cmake` is in `build/ProjectNestor/` but the prefix path is not picked up on re-runs against a stale build directory.

## Suggested Fix
In the `_build-nestor` recipe in `justfile`, ensure `conan install` always runs before `cmake -S ... -B ...`, or add `-DCMAKE_PREFIX_PATH="{{BUILD_ROOT}}/ProjectNestor"` to the cmake invocation to match the pattern used by Agamemnon.

## Phase
Phase 2 (Build) — e2e-walkthrough-report.md Issue #2

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -25,6 +25,15 @@ jobs:
           if [ "${{ matrix.compiler }}" = "clang" ]; then
             sudo apt-get install -y clang
           fi
+          pip install conan
+          conan profile detect --force
+      - name: Install Conan dependencies
+        env:
+          BUILD_TYPE: ${{ matrix.build_type }}
+        run: |
+          BUILD_TYPE_CAP=$(echo "$BUILD_TYPE" | sed 's/./\u&/')
+          conan install . --build=missing -s build_type="$BUILD_TYPE_CAP" \
+            --output-folder "build/$BUILD_TYPE"
       - name: Configure
         env:
           COMPILER: ${{ matrix.compiler }}

--- a/.github/workflows/code-coverage.yml
+++ b/.github/workflows/code-coverage.yml
@@ -13,7 +13,14 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - name: Install dependencies
-        run: sudo apt-get install -y ninja-build gcovr
+        run: |
+          sudo apt-get install -y ninja-build gcovr
+          pip install conan
+          conan profile detect --force
+      - name: Install Conan dependencies
+        run: |
+          conan install . --build=missing -s build_type=Debug \
+            --output-folder build/coverage
       - name: Configure
         run: cmake --preset coverage
       - name: Build

--- a/.github/workflows/static-analysis.yml
+++ b/.github/workflows/static-analysis.yml
@@ -23,7 +23,14 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - name: Install dependencies
-        run: sudo apt-get install -y ninja-build clang clang-tidy
+        run: |
+          sudo apt-get install -y ninja-build clang clang-tidy
+          pip install conan
+          conan profile detect --force
+      - name: Install Conan dependencies
+        run: |
+          conan install . --build=missing -s build_type=Debug \
+            --output-folder build/debug
       - name: Configure
         run: cmake --preset debug -DProjectNestor_ENABLE_CLANG_TIDY=ON
       - name: Build

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,9 +50,10 @@ FetchContent_Declare(
 
 set(NATS_BUILD_STREAMING OFF CACHE BOOL "" FORCE)
 set(NATS_BUILD_EXAMPLES  OFF CACHE BOOL "" FORCE)
+set(BUILD_TESTING_SAVED "${BUILD_TESTING}")
 set(BUILD_TESTING        OFF CACHE BOOL "" FORCE)
-
 FetchContent_MakeAvailable(natsc)
+set(BUILD_TESTING "${BUILD_TESTING_SAVED}" CACHE BOOL "" FORCE)
 
 # ─── Testable Core Library (everything except main) ────────────────────────
 add_library(${PROJECT_NAME}_core STATIC

